src/future.cpp

```diff
@@ -240,6 +240,14 @@ namespace detail {
     good since we need a concrete interface at this level. Above code could be simpler.
 */
 
+void async_(std::chrono::milliseconds delay, function<void()> f) {
+    async_([_delay = delay, _f = move(f)]{
+        if (_delay > std::chrono::milliseconds(0))
+        std::this_thread::sleep_for(_delay);
```

Comment:
Although this is better than an absolute block, this is going to block all tasks from executing on this thread (which is likely one of the pool threads) until the timer executes. On Windows this should become a wait callback object (see CreateThreadpoolWait). There are similar facilities for the other platforms, and we need to write a proper wait scheduler for the portable system.